core/input/input_event.cpp

```diff
@@ -436,6 +445,23 @@ String InputEventKey::as_text_key_label() const {
 	return mods_text.is_empty() ? kc : mods_text + "+" + kc;
 }
 
+String InputEventKey::as_text_location() const {
+	String loc;
+
+	switch (location) {
+		case KeyLocation::LEFT:
+			loc = "left";
+			break;
+		case KeyLocation::RIGHT:
+			loc = "right";
+			break;
+		default:
+			break;
```

Comment:
I remember making this decision, but I don't exactly remember the reasoning at the time. :thinking: 

I think it was flavoured by the other `as_text_*` functions returning (IIUC) an empty string when something is unknown (eg. [returning the value after checking `is_empty()`](https://github.com/godotengine/godot/pull/80231/files#diff-bd2f150d9c6b41b4bcce9ee39e6645d62ef2d2bd678d71ded8148a7bb513f691R478)):
```c++
if (kc.is_empty()) {
	return kc;
}
```

Though I notice now that there is currently also [similar logic for mods](https://github.com/godotengine/godot/pull/80231/files#diff-bd2f150d9c6b41b4bcce9ee39e6645d62ef2d2bd678d71ded8148a7bb513f691R509) in `to_string`:
```c++
String mods = InputEventWithModifiers::as_text();
mods = mods.is_empty() ? "none" : mods;
```

Which makes me realise that these functions are used to provide textual descriptions of the keypresses, and in that context the location is also a kind of a modifier. You have "Shift" or "Left Shift", just as you have "Z" or "Ctrl+Z".
"Hold Unknown Shift to run" just doesn't quite have the same ring to it! :wink: 

So I've (re-)convinced myself that returning an empty string is right for this function, but I'm not going to start a fight about it if you still think otherwise! :smile:

Reply:
> Which makes me realise that these functions are used to provide textual descriptions of the keypresses, and in that context the location is also a kind of a modifier. You have "Shift" or "Left Shift", just as you have "Z" or "Ctrl+Z".
"Hold Unknown Shift to run" just doesn't quite have the same ring to it! 😉

That makes sense in that context, so I'm good with the current logic!